fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FoldConstantRuleOnFE.java

```diff
@@ -515,6 +519,22 @@ public Expression visitCast(Cast cast, ExpressionRewriteContext context) {
         }
     }
 
+    // Check if the given literal value is safe to cast to the targetType.
+    // We need to guarantee FE cast result is identical with BE cast result.
+    // Otherwise, it's not safe.
+    protected boolean safeToCast(Cast cast) {
+        if (cast == null || cast.child() == null || cast.getDataType() == null) {
+            return true;
+        }
+        // Check double type.
+        if (cast.child() instanceof DoubleLiteral && cast.getDataType().isStringLikeType()) {
+            Double value = ((DoubleLiteral) cast.child()).getValue();
+            return -1E16 < value && value < 1E16;
```

Comment:
if value is nan or inf, this function will return false?

Reply:
Yes, it would return false. But I will change it to true so that the behavior would be the same with be.